src/BSON/UTCDateTime.c

```diff
@@ -275,6 +276,45 @@ PHP_METHOD(UTCDateTime, toDateTime)
 }
 /* }}} */
 
+/* {{{ proto array UTCDateTime::jsonSerialize()
+*/
+PHP_METHOD(UTCDateTime, jsonSerialize)
+{
+	php_phongo_utcdatetime_t *intern;
+	char                      s_milliseconds[24];
+	int                       s_milliseconds_len;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_UTCDATETIME_OBJ_P(getThis());
+
+	s_milliseconds_len = snprintf(s_milliseconds, sizeof(s_milliseconds), "%" PRId64, intern->milliseconds);
+
+	array_init_size(return_value, 1);
+
+#if PHP_VERSION_ID >= 70000
+	{
+		zval udt;
+
+		array_init_size(&udt, 1);
+		ADD_ASSOC_STRINGL(&udt, "$numberLong", s_milliseconds, s_milliseconds_len);
+		ADD_ASSOC_ZVAL_EX(return_value, "$date", &udt);
```

Comment:
The Extended JSON spec currently states that a UTCDateTime should yield:

```
{"$date": {"$numberLong": "<ms since epoch as an integer>"}}

Alternative, optional format:

{"$date": {"$isoDate": <ISO-8601 date string in UTC as a string>}}
```

I'm opting to use the first format as ISO-8601 has issues representing the full range of dates supported by the BSON date type (as discussed during the previous driver team meeting). Additionally, the `$numberLong` format is much easier to generate.

HHVM currently yields the `$isoDate` format, which I think we should change. Discussion is ongoing in the spec about whether drivers should always yield `$numberLong` and optionally provide a `$isoDate` purely for user readability purposes (i.e. it will never be parsed for the authoritative value).